### PR TITLE
Problem: Missing libc++ built for linux (fix #176)

### LINKED ISSUE #176: Problem: Missing libc++ built for linux
We are using libstdc++ to build play-cpp-sdk, while unreal use libc++. 

Link error will happen if use the sdk in unreal, check the same issue reported here: https://forums.unrealengine.com/t/linker-error-with-c-std-lib-on-linux/399040

Solution: 
Build play-cpp-sdk with clang and libc++ in linux

### PR DESCRIPTION
**Pull request by elecming**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

Close:
- https://github.com/cronos-labs/play-cpp-sdk/issues/176

Solutions:
- Support both g++ and clang++ on mac/linux
- Add libc++ support for linux

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-com/play-cpp-sdk/blob/main/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest main? 
- [ ] Have you checked your code compiles? (`cargo build`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`cargo test`)
- [ ] Have you checked your code formatting is correct? (`cargo fmt -- --check --color=auto`)
- [ ] Have you checked your basic code style is fine? (`cargo clippy`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`cargo audit`)
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-com/play-cpp-sdk/blob/main/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-com/play-cpp-sdk/blob/main/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-com/play-cpp-sdk/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

**Review by leejw51crypto — APPROVED** [review]
lgtm

**Comment by leejw51crypto** [comment]
removing draft?

**Comment by leejw51crypto** [comment]
is it clang?

**Comment by elecming** [comment]
@leejw51crypto it is not ready. Now both libraries libstdc++ and libc++ are used.. libstdc++ is needed to be removed.

**Comment by leejw51crypto** [comment]
if possible, this option can be const, so user easily can change?
clang or gcc

File: .github/workflows/linux-libc++-build.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          cd install
+          tar zcvf ../play_cpp_sdk_libc++_${PLATFORM}.tar.gz *
+          cd ..
+          sha256sum *.tar.gz > "checksums-$PLATFORM.txt"
+          echo "release_file=play_cpp_sdk_libc++_$PLATFORM.tar.gz" >> $GITHUB_ENV
+          echo "checksum_file=checksums-$PLATFORM.txt" >> $GITHUB_ENV
+          cat $GITHUB_ENV
+
+      - name: Upload binaries and bindings to Release
+        uses: softprops/action-gh-release@v1
```
Comment (tomtau) [inline .github/workflows/linux-libc++-build.yml:46]: not sure if this one needs extra (write) permissions on the token

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -38,9 +38,17 @@ jobs:
           name: debug-files
           path: debug_files.tar.gz
           if-no-files-found: ignore
+
+      - name: Build play-cpp-sdk library
+        run: CXX=clang++-10 CXXFLAGS=-stdlib=libc++ cargo build --package play-cpp-sdk --release
+
+      - name: Build demo project
+        working-directory: demo
+        run: make CXX=clang++-10
```
Comment (tomtau) [inline .github/workflows/test.yml:47]: any particular reason why clang 10, and not latest stable (14)?

**Review by tomtau — APPROVED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (elecming) [reply .github/workflows/test.yml:47]: Default clang is 11, headers were missing. I check under `/usr/include/c++` folder, only 9 and 10 are available:
image

Comment (tomtau) [reply .github/workflows/test.yml:47]: oh ok on ubuntu 20.04

File: .github/workflows/linux-libc++-build.yml (hunk 1 above)
Comment (elecming) [reply .github/workflows/linux-libc++-build.yml:46]: It seems it is fine. Same as before, after tagging, file will be uploaded as a draft release.

**Pull request merged**